File: Examples/test-suite/perl5/argcargvtest_runme.pl
use strict;
use warnings;
use Test::More tests => 17;
BEGIN { use_ok('argcargvtest') }
require_ok('argcargvtest');

my $largs = ["hi", "hola", "hello"];
is(argcargvtest::mainc($largs), 3, "test main typemap 1");

my $targs = ["hi", "hola"];
is(argcargvtest::mainv($targs, 0), "hi", "test main typemap 2a");
is(argcargvtest::mainv($targs, 1), "hola", "test main typemap 2b");
is(argcargvtest::mainv($targs, 2), "<<NULL>>", "test main typemap 2c");

my $errorVal = 0;
my $ret = eval qq(argcargvtest::mainv("hello", 1); \$errorVal = 1;);
is($ret, undef, "test main typemap 3");
is($errorVal, 0, "test main typemap 4");

is(argcargvtest::initializeApp($largs), undef, "test main typemap 5");

# Check that an empty array works.
my @empty_args = ();
is(argcargvtest::mainc(\@empty_args), 0, "test main typemap 6");
is(argcargvtest::mainv(\@empty_args, 0), "<<NULL>>", "test main typemap 6a");

# Check that empty strings are handled.
my @empty_string = ("hello", "", "world");
is(argcargvtest::mainc(\@empty_string), 3, "test main typemap 7");
is(argcargvtest::mainv(\@empty_string, 0), "hello", "test main typemap 8a");
is(argcargvtest::mainv(\@empty_string, 1), "", "test main typemap 8b");
is(argcargvtest::mainv(\@empty_string, 2), "world", "test main typemap 8c");
is(argcargvtest::mainv(\@empty_string, 3), "<<NULL>>", "test main typemap 8d");

ok(1, "done");

File: Examples/test-suite/octave/argcargvtest_runme.m
argcargvtest

largs={'hi','hola','hello'};
if (mainc(largs) != 3)
  error("bad main typemap");
endif

targs={'hi','hola'};
if (mainv(targs,1) != 'hola')
  error("bad main typemap");
endif

targs={'hi', 'hola'};
if (mainv(targs,0) != 'hi')
  error("bad main typemap");
endif
if (mainv(targs,1) != 'hola')
  error("bad main typemap");
endif
if (mainv(targs,2) != '<<NULL>>')
  error("bad main typemap");
endif

try
  error_flag = 0;
  mainv('hello',1);
  error_flag = 1;
catch
end_try_catch
if (error_flag)
  error("bad main typemap")
endif


initializeApp(largs);

# Check that an empty array works.
empty_args={};
if (mainc(empty_args) != 0)
  error("bad main typemap");
endif
if (mainv(empty_args,0) != '<<NULL>>')
  error("bad main typemap");
endif

# Check that empty strings are handled.
empty_string={"hello", blanks(0), "world"};
if (mainc(empty_string) != 3)
  error("bad main typemap");
endif
if (mainv(empty_string, 0) != "hello")
  error("bad main typemap");
endif
if (mainv(empty_string, 1) != "")
  error("bad main typemap");
endif
if (mainv(empty_string, 2) != "world")
  error("bad main typemap");
endif
if (mainv(empty_string, 3) != "<<NULL>>")
  error("bad main typemap");
endif
